README.md

```diff
@@ -2,6 +2,68 @@
 This smart contract provides a way for group members to assert, on chain, their intention to become a member of a
 [Provenance Blockchain Group](https://docs.cosmos.network/main/modules/group).
 
+This solves an issue in the groups module: any admin of any group can add a member to a group without their knowledge.  
+In order to create an ecosystem that respects group member involvement, this grants group members the ability to 
+explicitly state that they intended to become a member of a group.
+
+## Contract Instantiation
+
+To instantiate a new instance of the contract, the following parameters are required:
+
+* `contract_name`: This name will identify the contract when separate instances are deployed on chain.  It can be found
+via the `query_contract_state` query.
+* `attribute_name`: The [Provenance Blockchain Name Module](https://docs.provenance.io/modules/name-module) name that 
+will be bound to the contract.  When the `approve_group_membership` execution route is invoked, a
+[Provenance Blockchain Attribute](https://docs.provenance.io/modules/account) will be bound to the invoking account that
+includes the specified group id as its INT value.
+* `bind_attribute_name`: If specified as `true`, the value specified in `attribute_name` will be automatically bound 
+as a Provenance Blockchain Name to the contract during the instantiation process.  If this is omitted, the same name
+specified in `attribute_name` must manually be bound after the contract is instantiated.  This is useful in circumstances
+where the name will be bound using a restricted name module namespace.
+
+Example instantiation payload:
+```json
+{
+  "contract_name": "Sample identifying name for auxiliary contract",
+  "attribute_name": "somename.sc.pb",
+  "bind_attribute_name": true
+}
+```
+
+## Contract Execution
+
+In order for a member to verify their membership in a group, they must simply invoke the contract's sole execution route
+with the following payload:
```

Comment:
I know it is obvious, but maybe some sort of mention of 'with the appropriate group id'
```suggestion
with the following payload (containing the appropriate group id):
```